playbooks/services.yml

```diff
@@ -9,12 +9,12 @@
   roles:
     - role: helm_install
       tags: ['helm']
-    - role: kube_nginx_ingress
-      tags: ['ingress']
     - role: etcd_prometheus_exporter_cert
       tags: ['kube-prometheus']
     - role: kube_prometheus
       tags: ['kube-prometheus']
+    - role: kube_nginx_ingress
```

Comment:
If I'm not mistaken, `prometheus-operator` uses a `Job` to register the CRD. Since we can't really know when this `Job` finished, couldn't this result in a timing problem?

Reply:
Oh wasn't aware of that. So now even the doc of `prometheus-operator` is wrong stating:
```
helm install coreos/prometheus-operator --name prometheus-operator --namespace monitoring
helm install coreos/kube-prometheus --name kube-prometheus --namespace monitoring
```
There is no guarantee that the second line could execute correctly...

Reply:
And might explain some transient errors when installing kube-prometheus

Reply:
Suggestion: add a role "wait_prometheus_crd", which does (with retry on failure):

```
kubectl get customresourcedefinitions alertmanagers.monitoring.coreos.com prometheuses.monitoring.coreos.com servicemonitors.monitoring.coreos.com
```
(Taken from the helm test get-crd-job.yml of kube-prometheus)